tests/integration/tls_tests/test_tls.py

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env python3
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import pytest
+from pytest_operator.plugin import OpsTest
+
+from tests.integration.tls_tests.helpers import check_tls
+
+TLS_CERTIFICATES_APP_NAME = "tls-certificates-operator"
+DATABASE_APP_NAME = "mongodb"
+
+
+@pytest.mark.abort_on_fail
+async def test_build_and_deploy(ops_test: OpsTest) -> None:
+    """Build and deploy one unit of MongoDB and one unit of TLS."""
+    my_charm = await ops_test.build_charm(".")
+    await ops_test.model.deploy(my_charm)
+    await ops_test.model.wait_for_idle()
+
+    config = {"generate-self-signed-certificates": "true", "ca-common-name": "Test CA"}
+    await ops_test.model.deploy(TLS_CERTIFICATES_APP_NAME, channel="edge", config=config)
+    await ops_test.model.wait_for_idle(
+        apps=[TLS_CERTIFICATES_APP_NAME], status="active", timeout=1000
+    )
+
+
+async def test_enable_tls(ops_test: OpsTest) -> None:
+    """Verify each unit has TLS enabled after relating to the TLS application."""
+    # Relate it to the PostgreSQL to enable TLS.
+    await ops_test.model.relate(DATABASE_APP_NAME, TLS_CERTIFICATES_APP_NAME)
+    await ops_test.model.wait_for_idle(status="active", timeout=1000)
+
+    # Wait for all units enabling TLS.
+    for unit in ops_test.model.applications[DATABASE_APP_NAME].units:
+        assert await check_tls(ops_test, unit, enabled=True)
+
+
+async def test_disable_tls(ops_test: OpsTest) -> None:
```

Comment:
Would it be worth sending & receiving some data, and verifying that TLS is applied?

Reply:
`check_tls` sends a command to mongod and then receives the results of the command to verifies that TLS is applied.

I could send data, but I think this is sufficient